source/common/router/shadow_writer_impl.cc

```diff
@@ -6,18 +6,21 @@
 #include "common/common/assert.h"
 #include "common/http/headers.h"
 
+#include "absl/strings/str_join.h"
+
 namespace Envoy {
 namespace Router {
 
 void ShadowWriterImpl::shadow(const std::string& cluster, Http::MessagePtr&& request,
                               std::chrono::milliseconds timeout) {
-  // Switch authority to add a shadow postfix. This allows upstream logging to make a more sense.
-  // TODO PERF: Avoid copy.
-  std::string host = request->headers().Host()->value().c_str();
-  ASSERT(!host.empty());
-  host += "-shadow";
-  request->headers().Host()->value(host);
-
+  ASSERT(!request->headers().Host()->value().empty());
+  // Switch authority to add a shadow postfix. This allows upstream logging to make a more
```

Comment:
nit: Do you mind fixing the second sentence to not have 'a' in 'to make a more'.

Reply:
👍